scripts/analyze_deps.py

```diff
@@ -0,0 +1,250 @@
+                pkg_info = read_pkg_info_bytes(whl.read(whl.dist_info_path + '/METADATA'))
+                lib_name = pkg_info.get('Name')
+                requires = pkg_info.get_all('Requires-Dist')
+                for req in requires:
+                    parsed = requires_dist_re.match(req.split(';')[0].strip())
+                    req_name, spec = parsed.group('name', 'spec')
+                    spec = ','.join(sorted(spec.split(','))) if spec else ''
+                    if not req_name in dependencies:
+                        dependencies[req_name] = {}
+                    if not spec in dependencies[req_name]:
+                        dependencies[req_name][spec] = []
+                    dependencies[req_name][spec].append(lib_name)
+        except:
+            print('Failed to parse METADATA from %s' % (whl_path))
+    return dependencies
+
+def parse_setup(setup_filename):
+    mock_setup = textwrap.dedent('''\
+    def setup(*args, **kwargs):
+        __setup_calls__.append((args, kwargs))
+    ''')
+    parsed_mock_setup = ast.parse(mock_setup, filename=setup_filename)
+    with open(setup_filename, 'rt') as setup_file:
+        parsed = ast.parse(setup_file.read())
+        for index, node in enumerate(parsed.body[:]):
+            if (
+                not isinstance(node, ast.Expr) or
+                not isinstance(node.value, ast.Call) or
+                not hasattr(node.value.func, 'id') or
+                node.value.func.id != 'setup'
+            ):
+                continue
+            parsed.body[index:index] = parsed_mock_setup.body
+            break
+    
+    fixed = ast.fix_missing_locations(parsed)
+    codeobj = compile(fixed, setup_filename, 'exec')
+    local_vars = {}
+    global_vars = {'__setup_calls__': []}
+    current_dir = os.getcwd()
+    working_dir = os.path.dirname(setup_filename)
+    os.chdir(working_dir)
+    exec(codeobj, global_vars, local_vars)
+    os.chdir(current_dir)
+    _, kwargs = global_vars['__setup_calls__'][0]
+
+    requires = []
+    if 'install_requires' in kwargs:
+        requires += kwargs['install_requires']
+    if 'extras_require' in kwargs:
+        for extra in kwargs['extras_require'].values():
+            requires += extra
+    return requires
+
+def dict_compare(d1, d2):
+    d1_keys = set(d1.keys())
+    d2_keys = set(d2.keys())
+    intersect_keys = d1_keys.intersection(d2_keys)
+    added = d1_keys - d2_keys
+    removed = d2_keys - d1_keys
+    modified = {o : (d1[o], d2[o]) for o in intersect_keys if d1[o] != d2[o]}
+    return added, removed, modified
+
+class Logger(object):
+    def __init__(self, path):
+        self.terminal = sys.stdout
+        self.log = open(path, 'a')
+    
+    def write(self, message):
+        self.terminal.write(message)
+        self.log.write(message)
+    
+    def flush(self):
+        pass
+
+if __name__ == '__main__':
+    base_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
```

Comment:
In other python scripts, I usually see argparse being used instead of directly using sys.argv. In general, is this good policy, or is it more encouraged to only be used when special functionality of the arguments (keyword vs positional, accumulators, etc)?

Reply:
Oh, definitely a better approach. This was entirely laziness, and because when I started I had no args to parse, and slowly accumulated a few over time. I should clean this up before merge, if for no other reason than there's no --help to explain what args are supported.

Reply:
Fixed in 88c039a